subprojects/tooling-api/src/main/java/org/gradle/tooling/LongRunningOperation.java

```diff
@@ -120,6 +121,21 @@ public interface LongRunningOperation {
      */
     LongRunningOperation addJvmArguments(String... jvmArguments);
 
+    /**
+     * Sets additional system properties for the build.
+     * <p>
+     * This method has "setter" behavior, so the last invocation will overwrite previously set values.
```

Comment:
This is still a little confusing exactly what "additional system properties" mean. 

- How does this interact with System properties defined in the build (gradle.properties)? I think properties passed here should overwrite, but I don't know that they do.
- There are still immutable system properties that are propagated.

Since this says "additional system properties", someone might expect that we're still propagating all system properties from the client. 

Maybe we can say something like:

> Sets system properties to pass to the the build.
> By default, the Tooling API passes all system properties defined in the client to the build.
> If called, this method limits the system properties that are passed to the build, except for immutable system properties that need to match on both sides.